cassandra-4/src/main/java/io/debezium/connector/cassandra/Cassandra4CommitLogProcessor.java

```diff
@@ -246,38 +254,67 @@ private ProcessingResult callInternal() {
 
             try {
                 parseIndexFile();
-
                 while (!commitLog.completed) {
+                    LOGGER.info("Polling for completeness of idx file for: {}", commitLog.toString());
                     if (completePrematurely) {
                         LOGGER.info("{} completed prematurely", commitLog.toString());
                         return new ProcessingResult(commitLog, ProcessingResult.Result.COMPLETED_PREMATURELY);
                     }
-                    // TODO make this configurable maybe
+
+                    CommitLogPosition commitLogPosition = null;
+                    if ((segmentToOffset.get(commitLog.commitLogSegmentId) == null)) {
+                        LOGGER.info("Start to read the partial file : {}", commitLog.toString());
+                        commitLogPosition = new CommitLogPosition(commitLog.commitLogSegmentId, 0);
+                    }
+                    else if (segmentToOffset.get(commitLog.commitLogSegmentId) < commitLog.offsetOfEndOfLastWrittenCDCMutation) {
+                        LOGGER.info("Resume to read the partial file : {}", commitLog.toString());
+                        commitLogPosition = new CommitLogPosition(commitLog.commitLogSegmentId, segmentToOffset.get(commitLog.commitLogSegmentId));
+                    }
+                    else {
+                        LOGGER.info("No movement in HWM in IDX file: {}", commitLog.toString());
```

Comment:
what does HWM stands for?

Reply:
HWM stands for High Watermark, I understand and will update the log statement as HWM is not appropriate

Reply:
addressed